diff --git a/cicecore/shared/ice_distribution.F90 b/cicecore/shared/ice_distribution.F90
old mode 100644
new mode 100755
index 0f3f6c1..6e06069
--- a/cicecore/shared/ice_distribution.F90
+++ b/cicecore/shared/ice_distribution.F90
@@ -41,7 +41,8 @@ module ice_distribution
              ice_distributionGet,         &
              ice_distributionGetBlockLoc, &
              ice_distributionGetBlockID, &
-             create_local_block_ids
+             create_local_block_ids, &
+             proc_decomposition 
 
    character (char_len), public :: &
        processor_shape       ! 'square-pop' (approx) POP default config
