diff --git a/cicecore/cicedyn/general/ice_init.F90 b/cicecore/cicedyn/general/ice_init.F90
index 24ac40db..4ad296aa 100644
--- a/cicecore/cicedyn/general/ice_init.F90
+++ b/cicecore/cicedyn/general/ice_init.F90
@@ -872,7 +872,7 @@ subroutine input_data
       ! runid and runtype are obtained from the driver, not from the namelist
 
       if (my_task == master_task) then
-         history_file  = trim(runid) // ".cice" // trim(inst_suffix) //".h"
+         history_file  = trim(runid) // ".cice" // trim(inst_suffix)
          restart_file  = trim(runid) // ".cice" // trim(inst_suffix) //".r"
          incond_file   = trim(runid) // ".cice" // trim(inst_suffix) //".i"
          ! Note by tcraig - this if test is needed because the nuopc cap sets
