diff --git a/MOM6/src/framework/MOM_coupler_types.F90 b/MOM6/src/framework/MOM_coupler_types.F90.new
index f87b409..124d786 100644
--- a/MOM6/src/framework/MOM_coupler_types.F90
+++ b/MOM6/src/framework/MOM_coupler_types.F90.new
@@ -8,7 +8,10 @@ use MOM_couplertype_infra, only : CT_set_diags, CT_send_data, CT_write_chksums,
 use MOM_couplertype_infra, only : CT_copy_data, CT_increment_data, CT_rescale_data
 use MOM_couplertype_infra, only : CT_set_data, CT_extract_data, CT_redistribute_data
 use MOM_couplertype_infra, only : coupler_1d_bc_type, coupler_2d_bc_type, coupler_3d_bc_type
-use MOM_couplertype_infra, only : ind_flux, ind_alpha, ind_csurf
+use MOM_couplertype_infra, only : ind_flux, ind_deltap, ind_kw, ind_flux0
+use MOM_couplertype_infra, only : ind_pcair, ind_u10, ind_psurf
+use MOM_couplertype_infra, only : ind_alpha, ind_csurf, ind_sc_no
+use MOM_couplertype_infra, only : ind_runoff, ind_deposition
 use MOM_domain_infra,      only : domain2D
 use MOM_time_manager,      only : time_type
 
@@ -22,7 +25,10 @@ public :: atmos_ocn_coupler_flux, coupler_type_data_override
 public :: coupler_1d_bc_type, coupler_2d_bc_type, coupler_3d_bc_type
 ! These are encoding constant parameters that indicate whether a flux, solubility or
 ! surface ocean concentration are being set or accessed with an inquiry.
-public :: ind_flux, ind_alpha, ind_csurf
+public :: ind_flux, ind_deltap, ind_kw, ind_flux0
+public :: ind_pcair, ind_u10, ind_psurf
+public :: ind_alpha, ind_csurf, ind_sc_no
+public :: ind_runoff, ind_deposition
 
 !> This is the interface to spawn one coupler_bc_type into another.
 interface coupler_type_spawn
