diff --git a/MOM6/src/diagnostics/MOM_PointAccel.F90 b/MOM6/src/diagnostics/MOM_PointAccel.F90.new
index 30f0803..7f91f79 100644
--- a/MOM6/src/diagnostics/MOM_PointAccel.F90
+++ b/MOM6/src/diagnostics/MOM_PointAccel.F90.new
@@ -124,7 +124,7 @@ subroutine write_u_accel(I, j, um, hin, ADp, CDp, dt, G, GV, US, CS, vel_rpt, st
     if (CS%u_file == -1) then
       if (len_trim(CS%u_trunc_file) < 1) return
       call open_ASCII_file(CS%u_file, trim(CS%u_trunc_file), action=APPEND_FILE, &
-                           threading=MULTIPLE, fileset=SINGLE_FILE)
+                           threading=MULTIPLE, fileset=MULTIPLE)
       if (CS%u_file == -1) then
         call MOM_error(NOTE, 'Unable to open file '//trim(CS%u_trunc_file)//'.')
         return
@@ -465,7 +465,7 @@ subroutine write_v_accel(i, J, vm, hin, ADp, CDp, dt, G, GV, US, CS, vel_rpt, st
     if (CS%v_file == -1) then
       if (len_trim(CS%v_trunc_file) < 1) return
       call open_ASCII_file(CS%v_file, trim(CS%v_trunc_file), action=APPEND_FILE, &
-                           threading=MULTIPLE, fileset=SINGLE_FILE)
+                           threading=MULTIPLE, fileset=MULTIPLE)
       if (CS%v_file == -1) then
         call MOM_error(NOTE, 'Unable to open file '//trim(CS%v_trunc_file)//'.')
         return
