--- CDEPS/datm/atm_comp_nuopc.F90	2023-08-31 10:49:23.508874000 +1000
+++ extra_sources/atm_comp_nuopc.F90	2023-08-31 10:48:55.428272000 +1000
@@ -52,6 +52,12 @@
   use datm_datamode_jra_mod     , only : datm_datamode_jra_restart_write
   use datm_datamode_jra_mod     , only : datm_datamode_jra_restart_read
 
+  use datm_datamode_jra55do_mod , only : datm_datamode_jra55do_advertise
+  use datm_datamode_jra55do_mod , only : datm_datamode_jra55do_init_pointers
+  use datm_datamode_jra55do_mod , only : datm_datamode_jra55do_advance
+  use datm_datamode_jra55do_mod , only : datm_datamode_jra55do_restart_write
+  use datm_datamode_jra55do_mod , only : datm_datamode_jra55do_restart_read
+
   use datm_datamode_clmncep_mod , only : datm_datamode_clmncep_advertise
   use datm_datamode_clmncep_mod , only : datm_datamode_clmncep_init_pointers
   use datm_datamode_clmncep_mod , only : datm_datamode_clmncep_advance
@@ -342,6 +348,7 @@
     if ( trim(datamode) == 'CORE2_NYF'    .or. &
          trim(datamode) == 'CORE2_IAF'    .or. &
          trim(datamode) == 'CORE_IAF_JRA' .or. &
+         trim(datamode) == 'JRA55do'.or. &
          trim(datamode) == 'CLMNCEP'      .or. &
          trim(datamode) == 'CPLHIST'      .or. &
          trim(datamode) == 'GEFS'         .or. &
@@ -361,6 +368,10 @@
        call datm_datamode_jra_advertise(exportState, fldsExport, flds_scalar_name, &
             flds_co2, flds_wiso, flds_presaero, flds_presndep, rc)
        if (ChkErr(rc,__LINE__,u_FILE_u)) return
+    case ('JRA55do')
+       call datm_datamode_jra55do_advertise(exportState, fldsExport, flds_scalar_name, &
+            flds_co2, flds_wiso, flds_presaero, flds_presndep, rc)
+       if (ChkErr(rc,__LINE__,u_FILE_u)) return
     case ('CLMNCEP')
        call datm_datamode_clmncep_advertise(exportState, fldsExport, flds_scalar_name, &
             flds_co2, flds_wiso, flds_presaero, flds_presndep, flds_preso3, rc)
@@ -606,6 +617,9 @@
        case('CORE_IAF_JRA')
           call datm_datamode_jra_init_pointers(exportState, sdat, rc)
           if (ChkErr(rc,__LINE__,u_FILE_u)) return
+       case('JRA55do')
+          call datm_datamode_jra55do_init_pointers(exportState, sdat, rc)
+          if (ChkErr(rc,__LINE__,u_FILE_u)) return
        case('CLMNCEP')
           call datm_datamode_clmncep_init_pointers(importState, exportState, sdat, rc)
           if (ChkErr(rc,__LINE__,u_FILE_u)) return
@@ -630,6 +644,8 @@
              call datm_datamode_core2_restart_read(restfilm, inst_suffix, logunit, my_task, mpicom, sdat)
           case('CORE_IAF_JRA')
              call datm_datamode_jra_restart_read(restfilm, inst_suffix, logunit, my_task, mpicom, sdat)
+          case('JRA55do')
+             call datm_datamode_jra55do_restart_read(restfilm, inst_suffix, logunit, my_task, mpicom, sdat)
           case('CLMNCEP')
              call datm_datamode_clmncep_restart_read(restfilm, inst_suffix, logunit, my_task, mpicom, sdat)
           case('CPLHIST')
@@ -677,6 +693,9 @@
     case('CORE_IAF_JRA')
        call datm_datamode_jra_advance(exportstate, target_ymd, target_tod, sdat%model_calendar, rc)
        if (ChkErr(rc,__LINE__,u_FILE_u)) return
+    case('JRA55do')
+       call datm_datamode_jra55do_advance(exportstate, target_ymd, target_tod, sdat%model_calendar, rc)
+       if (ChkErr(rc,__LINE__,u_FILE_u)) return       
     case('CLMNCEP')
        call datm_datamode_clmncep_advance(mainproc, logunit, mpicom,  rc)
        if (ChkErr(rc,__LINE__,u_FILE_u)) return
@@ -705,6 +724,9 @@
        case('CORE_IAF_JRA')
           call datm_datamode_jra_restart_write(case_name, inst_suffix, target_ymd, target_tod, &
                logunit, my_task, sdat)
+       case('JRA55do')
+          call datm_datamode_jra55do_restart_write(case_name, inst_suffix, target_ymd, target_tod, &
+               logunit, my_task, sdat)
        case('CLMNCEP')
           call datm_datamode_clmncep_restart_write(case_name, inst_suffix, target_ymd, target_tod, &
                logunit, my_task, sdat)
