diff --git a/dshr/dshr_mod.F90 b/dshr/dshr_mod.F90
index ca87f2e..7d6bd71 100644
--- a/dshr/dshr_mod.F90
+++ b/dshr/dshr_mod.F90
@@ -1058,7 +1058,6 @@ contains
     type(io_desc_t)   :: pio_iodesc
     integer           :: oldmode
     integer           :: rcode
-    character(len=CS) :: lrpfile
     character(*), parameter :: F00   = "('(dshr_restart_write) ',2a,2(i0,2x))"
     !-------------------------------------------------------------------------------
 
@@ -1071,7 +1070,7 @@ contains
 
     ! write restart info to rpointer file
     if (my_task == main_task) then
-       open(newunit=nu, file=trim(lrpfile), form='formatted')
+       open(newunit=nu, file=trim(rpfile), form='formatted')
        write(nu,'(a)') rest_file_model
        close(nu)
        write(logunit,F00)' writing ',trim(rest_file_model)
